tutorials/3d/using_decals.rst

```diff
@@ -0,0 +1,282 @@
+impacts and explosion scorches.
+
+Using an AnimationPlayer node or a script, decals can be made to fade over time
+(and then be removed using ``queue_free()``) to improve performance.
+
+Blob shadows
+^^^^^^^^^^^^
+
+Blob shadows are frequently used in mobile projects (or to follow a retro art
+style), as real-time lighting tends to be too expensive on low-end mobile
+devices. However, when relying on baked lightmaps with fully baked lights,
+dynamic objects will not cast *any* shadow from those lights. This makes dynamic
+objects in lightmapped scenes look flat in comparison to real-time lighting,
+with dynamic objects almost looking like they're floating.
+
+Thanks to blob shadows, dynamic objects can still cast an approximative shadow.
+Not only this helps with depth perception in the scene, but this can also be a
+gameplay element, especially in 3D platformers. The blob shadow's length can be
+extended to let the player know where they will land if they fall straight down.
+
+Even with real-time lighting, blob shadows can still be useful as a form of
+ambient occlusion for situations where SSAO is too expensive or too unstable due
+to its screen-space nature. For example, vehicles' underside shadows are
+well-represented using blob shadows.
+
+.. figure:: img/decals_blob_shadow.webp
+   :align: center
+   :alt: Blob shadow under object comparison
+
+   Blob shadow under object comparison
+
+Quick start guide
+-----------------
+
+Creating decals in the editor
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+1. Create a Decal node in the 3D editor.
+2. In the inspector, expand the **Textures** section and load a texture in
+   **Textures > Albedo**.
+3. Move the Decal node towards an object, then rotate it so the decal is visible
+   (and in the right orientation). If the decal appears mirrored, try to rotate
+   it by 180 degrees. You can double-check whether it's in the right orientation
+   by increasing **Parameters > Normal Fade** to 0.5. This will prevent the Decal
+   from being projected on surfaces that are not facing the decal.
+4. If your decal is meant to affect static objects only, configure it to prevent
+   affecting dynamic objects (or vice versa). To do so, change the decal's
+   **Cull Mask** property to exclude certain layers. After doing this, modify
+   your dynamic objects' MeshInstance3D nodes to change their visibility layers.
+   For instance, you can move them from layer 1 to layer 2, then disable layer 2
+   in the decal's **Cull Mask** property.
+
+Creating decals during gameplay
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Decal nodes can be created from a script like any other node. The main part that can
+be challenging is correctly rotating the decal so it can be visible without
+distortion.
+
+The following code snippet creates a decal at the current mouse cursor location
+when the space bar is pressed:
+
+.. TODO: Add code sample for rotating the decal based on the raycast collision normal.
```

Comment:
I couldn't figure out this part, but we should merge this documentation before 4.0 is released anyway.